bids/layout/layout.py

```diff
@@ -206,19 +206,26 @@ def __init__(self, root, validate=True, absolute_paths=True,
         self.sources = sources
         self.regex_search = regex_search
         self.config_filename = config_filename
+        # Store init arguments for saving
+        self._sanitize_init_args(
+            root=root, validate=validate, absolute_paths=absolute_paths,
+            derivatives=derivatives, ignore=ignore, force_index=force_index,
+            index_metadata=index_metadata, config=config)
 
         if database_file is not None:
```

Comment:
```suggestion
        if database_path is None and database_file is not None:
```

Reply:
This is an edge case, but we may as well check to make sure that `database_path` is also `None`... if for some inscrutable reason a user passes both, we don't want to overwrite `database_path`.